### ISSUE TITLE
Timed out dig lookups causes TCL error preventing the calling proc from continuing

### ISSUE THREAD

**Issue by Robby-** [aaronhurt/tcl-scripts#2]
I came across this when I saw Tornevall had some issues.

This kind of error (timed out connection, shown below) seems to block the calling proc from continuing any further if this call fails. In the circumstances that I use this script this is bad and continuance is wanted regardless of a timed out connection to some blacklist DNS server.

The error:

> Tcl error: Error calling dig:  (/usr/bin/dig 230.88.162.195.dnsbl.tornevall.org ANY): 
> Tcl error: ; <<>> DiG 9.5.2-RedHat-9.5.2-1.fc10 <<>> 230.88.162.195.dnsbl.tornevall.org ANY
> Tcl error: ;; global options:  printcmd
> Tcl error: ;; connection timed out; no servers could be reached
> Tcl error: child process exited abnormally

Luckily this timeout does not seem to prevent other blacklist DNS servers from being queried (but it of course slows it down), but this has me confused a bit because there are 2 other blacklists configured aswell, so why does it even throw this error unless for example ALL blacklists were unreachable due to DNS resolution issues on the network (which I simulated by blocking all outgoing port 53 traffic)?

Couldn't this be made so that no error is returned in the case of a timed out connection? Ie, only make it error for more serious problems with calling the dig binary (like for example failing to execute the binary because it is set to chmod 0000 or something)?

Also, when I was reproducing this error I saw that every blacklist lookup seems to hang the bot untill the error is returned by the ::rblchk::dig proc, some good 20 seconds later.

I wanted to quickly work around the problem by making it return 0 or 1 but that actually gave another problem: every configured blacklist lookup returned a positive result while in reality the IP was not listed at all so I had to undo that again.

Edit: I just overlooked a small detail in that error output, the error is due to the dig process not exiting cleanly due to this time out. (duh!)
Perhaps the workings of how the script checks for availability of the dig binary should be changed because current behavior is not really ideal since there's no real issue here, only a timed out lookup.

Sorry if I don't make sense somewhere, it's late in the morning and I've been testing/working this for a while. ;)

**Comment by aaronhurt** [comment]
Sorry for the really late reply, but I've been pretty swamped at work lately.  I have read your issue and think I have a solution for it but I probably won't be able to commit it till at least this coming weekend.

**Comment by aaronhurt** [comment]
Fixed ... I still return an error for this because it IS an error.  The above actually doesn't have anything to do with tornevall.org ... it appears it is actually timing out because it cannot get a reply from any of your configured resolvers (/etc/resolve.conf).  I added all of the defined return codes for dig to the script to cleanup the output of known errors and reduced the dig timeout from the default of 5 seconds per query to 1 second.
